python interface hangs if you try to set a non-existent endpoint.

I am using the python interface interactively. If I try to use a `set` command on an endpoint that doesn't exist, then the process just hangs. To get out, I have to kill the python process violently, either by ctrl-z and then killing the process, or by killing the container.

```
In [1]: from dripline.core import Interface                
fail!
fail!

In [2]: auths_file = '/etc/rabbitmq-secret/authentications.json'                                                      

In [3]: the_interface = Interface(dripline_config={'auth-file': auths_files_file})                                    
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-3-64ce7b469bd6> in <module>
----> 1 the_interface = Interface(dripline_config={'auth-file': auths_files_file})

NameError: name 'auths_files_file' is not defined

In [4]: the_interface = Interface(dripline_config={'auth-file': auths_file})                                          

In [5]: the_interface.set('curved_mirror_set_position', 0) 
Out[5]: <_dripline.core.MsgReply at 0x7f0671debd30>

In [6]: the_interface.set('modemap_measurement_status', 'start_measurement')                                          
2020-06-15 17:24:15 [ERROR] /library/core.cc(282): Message was returned: Message returned. Reply code: 312 NO_ROUTE
^C


^C^C
^C^C^C^C^C^C

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.1)
-project( DriplinePy VERSION 4.4.5 )
+project( DriplinePy VERSION 4.4.6 )
 
 list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/dripline-cpp/scarab/cmake )
 include( PackageBuilder )

--- a/bin/dl-serve
+++ b/bin/dl-serve
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
 import datetime
+import importlib.util
 import inspect
 import traceback
 
@@ -38,7 +39,9 @@ class Serve:
         extra_namespace = object()
         if module_path:
             try:
-                extra_namespace = imp.load_source('extra_namespace', module_path)
+                spec = importlib.util.spec_from_file_location('extra_namespace', module_path)
+                extra_namespace = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(extra_namespace)
             except IOError as e:
                 logger.error("unable to load source from: {}".format(module_path))
                 raise e

--- a/chart/Chart.yaml
+++ b/chart/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 ## the appVersion is used as the container image tag for the main container in the pod from the deployment
 ## it can be overridden by a values.yaml file in image.tag
-appVersion: "v4.4.5-amd64"
+appVersion: "v4.4.6-amd64"
 description: Deploy a dripline-python microservice
 name: dripline-python
 version: 1.1.1

--- a/dripline/core/interface.py
+++ b/dripline/core/interface.py
@@ -2,7 +2,7 @@ __all__ = []
 
 import scarab
 
-from dripline.core import op_t, Core, DriplineConfig, Receiver, MsgRequest
+from dripline.core import op_t, Core, DriplineConfig, Receiver, MsgRequest, DriplineError
 
 import logging
 logger = logging.getLogger(__name__)
@@ -30,7 +30,10 @@ class Interface(Core):
         '''
         a_specifier = specifier if specifier is not None else ""
         a_request = MsgRequest.create(payload=scarab.to_param(payload), msg_op=msgop, routing_key=target, specifier=a_specifier)
-        return self.send(a_request)
+        receive_reply = self.send(a_request)
+        if not receive_reply.successful_send:
+            raise DriplineError('unable to send request')
+        return receive_reply
 
     def get(self, endpoint, specifier=None, timeout=0):
         '''

--- a/dripline/implementations/postgres_interface.py
+++ b/dripline/implementations/postgres_interface.py
@@ -142,7 +142,7 @@ class SQLTable(Endpoint):
             this_select = this_select.where(getattr(self.table.c,c)<v)
         for c,v in where_gt_dict.items():
             this_select = this_select.where(getattr(self.table.c,c)>v)
-        result = self.provider.engine.execute(this_select)
+        result = self.service.engine.execute(this_select)
         return (result.keys(), [i for i in result])
 
     def _insert_with_return(self, insert_kv_dict, return_col_names_list):

--- a/dripline/implementations/postgres_interface.py
+++ b/dripline/implementations/postgres_interface.py
@@ -24,7 +24,7 @@ from itertools import groupby
 import collections
 
 # local imports
-from dripline.core import Service, Endpoint
+from dripline.core import Service, Endpoint, ThrowReply
 
 import logging
 logger = logging.getLogger(__name__)

--- a/module_bindings/dripline_core/core_pybind.hh
+++ b/module_bindings/dripline_core/core_pybind.hh
@@ -20,6 +20,7 @@ namespace dripline_pybind
         pybind11::class_< dripline::sent_msg_pkg,
                           std::shared_ptr< dripline::sent_msg_pkg >
                         >( mod, "SentMessagePackage", "Data structure for sent messages" )
+            .def_property_readonly( "successful_send", [](const dripline::sent_msg_pkg& an_obj){ return an_obj.f_successful_send; } )
             ;
 
         all_items.push_back( "Core" );

